### PR TITLE
Api/get end point to fetch all historical data

### PR DESCRIPTION
**Pull request by BonfaceKilz**
# Description
This PR adds an extra end-point to fetch all the RIF metadata.  In addition, this PR also fixes the GET wiki endpoint to use the correct header to fetch json-ld data.

# Testing

```
╭─munyoki@saitama ~  
╰─➤  curl localhost:8080/api/metadata/wiki/1/history
[
  {
    "comment": "hippocampal CA1-3 expression signature", 
    "created": "2005-12-17T00:23:23", 
    "email": "rwilliam@nb.utmem.edu", 
    "pubmed_id": "15084669", 
    "species": "mouse", 
    "symbol": "Lpl", 
    "type": "gnc:GNWikiEntry", 
    "version": "0", 
    "wiki_identifier": "gn:wiki-1-0"
  }
]

curl localhost:8080/api/metadata/wiki/1/history --header "Accept: application/ld+json"

{
  "@context": {
    "categories": "gnt:belongsToCategory", 
    "comment": "rdfs:comment", 
    "created": "dct:created", 
    "data": "@graph", 
    "dct": "http://purl.org/dc/terms/", 
    "email": "foaf:mbox", 
    "foaf": "http://xmlns.com/foaf/0.1/", 
    "gn": "http://genenetwork.org/id/", 
    "gnc": "http://genenetwork.org/category/", 
    "gnt": "http://genenetwork.org/term/", 
    "id": "dct:identifier", 
    "initial": "gnt:initial", 
    "pubmed_id": "dct:references", 
    "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#>", 
    "rdfs": "http://www.w3.org/2000/01/rdf-schema#", 
    "reason": "gnt:reason", 
    "species": "gnt:species", 
    "symbol": "rdfs:label", 
    "type": "@type", 
    "version": "gnt:hasVersion", 
    "web_url": "foaf:homepage", 
    "wiki_identifier": "@id"
  }, 
  "data": [
    {
      "comment": "hippocampal CA1-3 expression signature", 
      "created": "2005-12-17T00:23:23", 
      "email": "rwilliam@nb.utmem.edu", 
      "pubmed_id": "15084669", 
      "species": "mouse", 
      "symbol": "Lpl", 
      "type": "gnc:GNWikiEntry", 
      "version": "0", 
      "wiki_identifier": "gn:wiki-1-0"
    }
  ]
}

```

### CODE REVIEW

**Comment by jnduli** [comment]
Here's what I expected the data section to contain:
```
{
  "categories": [
    "Behavior", 
    "Biochemistry"
  ], 
  "comment": "XXXX", 
  "email": "XXXX", 
  "initial": "XXXX", 
  "pubmed_ids": [
    "XXXX", 
    "XXXX"
  ], 
  "reason": "XXXX", 
  "species": "mouse", 
  "symbol": "XXXX", 
  "version": 14, 
  "weburl": "XXXX"
}
```
The diffs I see are:

- we don't have multiple pubmed_ids.
- we don't have categories, initial, weburl and reason
- version we return in a string instead of int
- we have an extra wiki_identifier(which I understand), but what's the purpose for type?

**Comment by BonfaceKilz** [comment]
@jnduli addressed above comments:

```
╭─munyoki@saitama ~/projects/genenetwork3  ‹api/get-end-point-to-fetch-all-historical-data*› 
╰─➤  curl localhost:8080/api/metadata/wiki/587/history 
[
  {
    "categories": [
      "Physiology and function", 
      "Epigenetic modulation", 
      "Expression patterns: temporal variation"
    ], 
    "comment": "Imprinted in mouse and human. Peg3 is a paternally expressed, zinc finger protein. Peg3 is expressed in early somites, branchial arches and other mesodermal tissues, as well as in the hypothalamus. Loss of Peg3 expression results in reduced maternal care, maternal food intake, milk let-down, placental size, impaired fetal growth, suckling and postnatal growth, and delayed puberty. from Morison", 
    "created": "2010-06-24T04:32:53", 
    "email": "rwilliam@nb.uthsc.edu", 
    "initial": "RWW", 
    "pubmed_ids": [
      7550314, 
      15306355
    ], 
    "reason": "more", 
    "species": "mouse", 
    "symbol": "Peg3", 
    "version": 0, 
    "web_url": "http://igc.otago.ac.nz/Search.html", 
    "wiki_identifier": "gn:wiki-587-0"
  }, 
  {
    "categories": [
      "Physiology and function", 
      "Epigenetic modulation", 
      "Expression patterns: temporal variation"
    ], 
    "comment": "Peg3 is a paternally expressed, zinc finger protein. Peg3 is expressed in early somites, branchial arches and other mesodermal tissues, as well as in the hypothalamus. Loss of Peg3 expression results in reduced maternal care, maternal food intake, milk let-down, placental size, impaired fetal growth, suckling and postnatal growth, and delayed puberty. \nGlasgow et al, 2005 have described a Peg3 antisense transcript in rat APeg3. Its imprinting status is not yet known.", 
    "created": "2006-10-31T20:42:43", 
    "email": "rwilliam@nb.utmem.edu", 
    "initial": "RWW", 
    "pubmed_ids": [
      15306355, 
      7550314
    ], 
    "reason": "", 
    "species": "mouse", 
    "symbol": "Peg3", 
    "version": 1, 
    "web_url": "", 
    "wiki_identifier": "gn:wiki-587-1"
  }
]
```

File: gn3/db/rdf/wiki.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            result["categories"] = [categories]
+        pmids = result.get("pubmed_ids")
+        if pmids and isinstance(pmids, str):
+            result["pubmed_ids"] = [pmids]
+        elif pmids:
+            result["pubmed_ids"] = [int(pmid) for pmid in pmids]
+        else:
+            result["pubmed_ids"] = []
+        result["version"] = int(result["version"])
+    # We manually sort the array, since somehow "ORDER BY" in the
```
Comment (jnduli) [inline gn3/db/rdf/wiki.py]: should we also mention this in the docs for the function in question?

File: gn3/db/rdf/wiki.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    # sparql query is jumbled up during framing and compacting.  This
+    # operation isn't heavy since we don't get many versions per wiki
+    # entry.
+    results["data"] = sorted(data, key=lambda d: d["created"], reverse=True)
```
Comment (jnduli) [inline gn3/db/rdf/wiki.py]: The `createtime` column in our db defaults to `0000-00-000`. I suggest we change the key to `(d['version'], d['created'])` instead

Comment (BonfaceKilz) [reply gn3/db/rdf/wiki.py]: I deleted this commit and did a force push.

File: gn3/db/rdf/wiki.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        else:
+            result["pubmed_ids"] = []
+        result["version"] = int(result["version"])
+
+    # We manually sort the array, since the SPARQL engine does not
+    # provide a guarantee that it will support an ORDER BY clause in a
+    # CONSTRUCT. Using ORDER BY on a solution sequence for a CONSTRUCT
+    # or DESCRIBE query has no direct effect because only SELECT
+    # returns a sequence of results.  See:
+    # <https://stackoverflow.com/questions/78186393/how-apply-an-order-to-construct-in-sparql#:~:text=SPARQL%20engine%20does%20respect%20such,is%20a%20set%20of%20triples.>
```
Comment (jnduli) [inline gn3/db/rdf/wiki.py]: can we replace this with: https://stackoverflow.com/q/78186393

Comment (BonfaceKilz) [reply gn3/db/rdf/wiki.py]: Done.

**Pull request merged**